File: nonrepeatingchar.py
""" Find the first non repeating character in a string. """

def firstNonRepeatingCharacter(string):
    # create dictionary to store character counts
    count_dict = {}

    # loop string and check if char is in the count dict, if not, add it
    # if it is already present, increment the count
    for char in string:
        if char in count_dict:
            count_dict[char] += 1
        else:
            count_dict[char] = 1
    # loop the keys in the count dict and check if the value is equal to 1,
    # then return the index of that char in the string
    for key in count_dict.keys():
        if count_dict[key] == 1:
            return string.index(key)
    return -1

File: nonrepeatingchar.js
/* Find the first non repeating character in a string. */

function firstNonRepeatingCharacter(string) {
    // create a count dictionary
    let count_dict = {}
    // loop characters in string, and check if it is in the dict
    // if not present, set count dict at that char to 1, else increment again
    for(let char of string){
      if (count_dict[char] == undefined){
        count_dict[char] = 1;
      }else{
        count_dict[char]++;
      }
    }
    // loop keys in count dict, find the key where the value is 1 and return
    // the index of that char in the string
    for(let key in count_dict){
      if (count_dict[key] == 1){
        return string.indexOf(key)
      }
    }
    
    return -1;
  }